test-site/src/components/MapPin.tsx

```diff
@@ -1,18 +1,55 @@
-import { PinComponent } from '@yext/search-ui-react';
+import { useEffect, useRef, useState } from 'react';
+import { Map, Popup, LngLatLike } from 'mapbox-gl';
+import { PinComponent, Coordinate } from '@yext/search-ui-react';
+import { Result } from '@yext/search-headless-react';
 import { Location } from '../pages/LocationsPage';
 
-export const MapPin: PinComponent<Location> = () => {
+export interface MapPinProps<T> {
+  coordinate: Coordinate,
+  index: number,
+  mapbox: Map,
+  result: Result<T>
+}
+
+const transformToMapboxCoord = (coordinate: Coordinate): LngLatLike => ({
+  lng: coordinate.longitude,
+  lat: coordinate.latitude,
+});
+
+export const MapPin: PinComponent<Location> = (props: MapPinProps<Location>) => {
+  const { coordinate, mapbox, result } = props;
+  const [active, setActive] = useState(false);
```

Comment:
instead of maintaining our own active state does [`Popup.isOpen`](https://docs.mapbox.com/mapbox-gl-js/api/markers/#popup#isopen) work?

Reply:
We could use that for the conditional in useEffect, but having the active state hook is necessary for triggering the useEffect. There doesn't seem to be any Popup event for triggering one to open, only a callback for when one is

Reply:
Instead of a useEffect we could just add the popup in the click handler right?

Reply:
I think that might be slightly simpler...?

Reply:
No that hasn't been able to work for me or @nmanu1